### PR TITLE
MINIFICPP-2273 Update minifi documentation

### PR DESCRIPTION
**Pull request by martinzink**
I've restructred minifi's documentation a little bit, but I mainly focused on minifi cpp side (thats what im familiar with).
There were two sites previously with very similar information.
https://nifi.apache.org/minifi/getting-started.html
https://nifi.apache.org/minifi/minifi-java-agent-quick-start.html

I've merged these together and moved the additional walkthrough to a separate page. (and removed any cpp specific information)

The getting-started page is now helps selecting the minifi flavour and links to the java or cpp quick start page.
I've also created a cpp specific quick start that links to the live documentation on github.

I feel like we should remove most of java docs aswell because they feel really out of date, but Im not that familiar with minifi java to do so.

### CODE REVIEW

File: source/minifi/faq.md [hunk 1]
```diff
@@ -9,3 +9,18 @@ containerEnabled: true
...
+However, it results in a larger binary distribution and consumes greater system resources.
+If you require maximum flexibility to make routing and processing decisions at your data's point of origin, the Java agent is a good fit.
+
+The C++ agent is a native reimplementation of MiNiFi Java.
+Due to the absence of a JVM, it consumes fewer system resources but has a [limited subset of processors](https://github.com/apache/nifi-minifi-cpp/blob/main/PROCESSORS.md).
+If your primary concern is gathering and pushing data to downstream consumers while minimizing system impact, the C++ agent is a good fit.
+
+### Can I use MiNiFi Toolkit Converter for MiNiFi C++?
+
+Unfortunately due to differences in naming, properties and processor availability in MiNiFi C++ and NiFi, the MiNiFi Toolkit Converter is not recommended for MiNiFi C++.
```
Comment (szaszm) [inline source/minifi/faq.md:26]:
```suggestion
Unfortunately due to differences in naming, properties and processor availability in MiNiFi C++ and NiFi, the MiNiFi Toolkit Converter is not recommended for MiNiFi C++. To create a flow definition for MiNiFi C++, check out our [examples](https://github.com/apache/nifi-minifi-cpp/tree/main/examples).
```

File: source/minifi/minifi-cpp-agent-quick-start.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- generation of data provenance
+- integration with NiFi for follow-on dataflow management and full chain of custody of information 
+
+Perspectives of the role of MiNiFi should be from the perspective of the agent acting immediately at, or directly adjacent to, source sensors, systems, or servers.
+
+- [Source code](https://github.com/apache/nifi-minifi-cpp)
+- [Downloads](download.html)
+- [Documentation](https://github.com/apache/nifi-minifi-cpp/blob/main/README.md)
+- [List of supported Processors](https://github.com/apache/nifi-minifi-cpp/blob/main/PROCESSORS.md)
+- [Examples](https://github.com/apache/nifi-minifi-cpp/blob/main/examples/README.md)
```
Comment (szaszm) [inline source/minifi/minifi-cpp-agent-quick-start.md:26]: Maybe this list could be extracted and shared between this file and getting-started.md? Having a single source of truth would reduce the likelihood of leaving one of the two lists outdated in the future.

File: source/minifi/faq.md [hunk 3]
```diff
@@ -9,3 +9,18 @@ containerEnabled: true
 ### How do you pronounce MiNiFi?
 
 "minify" (_min_-uh-fahy) is the preferred pronunciation.
+
+
+### What's the difference between MiNiFi Java and MiNiFi C++?
```
Comment (exceptionfactory) [inline source/minifi/faq.md:14]: I recommend removing the MiNiFi FAQ page and including this information in one of the other pages, such as the main MiNiFi description.

File: source/minifi/getting-started.md [hunk 4]
```diff
@@ -1,124 +1,42 @@
 ---
-title:     Apache NiFi MiNiFi Getting Started
```
Comment (exceptionfactory) [inline source/minifi/getting-started.md:2]: The Getting Started page does not have much in the way of additional detail. What do you think about removing the page completely and just linking to the README.md for MiNiFi and MiNiFi C++ on GitHub?

File: source/minifi/minifi-cpp-agent-quick-start.md (hunk 2 above)
Comment (exceptionfactory) [reply source/minifi/minifi-cpp-agent-quick-start.md:26]: I agree that consolidating this list and avoiding the creation of a new getting started page would be a better approach.

Comment (martinzink) [reply source/minifi/minifi-cpp-agent-quick-start.md]: I've redesigned this part, I've created instead a separate dropdown for minifi java and minifi cpp, the landing page tells you about what to choose and after that you can go to the appropriate pages using the dropdowns.

File: source/minifi/getting-started.md (hunk 4 above)
Comment (martinzink) [reply source/minifi/getting-started.md:2]: Good idea :+1: , I've moved this information onto the landing page and created separate dropdown for java and cpp in https://github.com/apache/nifi-site/pull/79/commits/396948ba0db6a691a0ece71fcf64bff8a75ebb7a

File: source/minifi/minifi-java-guide-pushing-nifi-minifi.md [hunk 5]
```diff
@@ -94,7 +50,7 @@ No validation errors found in template.
 ```
 
 Open the config.yml file and add Port and Host to a Properties section at the end of the Input Ports section (at the end of the file):
-```
+````
```
Comment (szaszm) [inline source/minifi/minifi-java-guide-pushing-nifi-minifi.md:53]: Is the 4th backtick intentional? Is it related to the single backtick at the end of the file?

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for making the updates @martinzink.

Looking at the latest version, I think it would be better to introducing the new about pages and simply have two separate sections on the main MiNiFi page, one paragraph for Java, and one for CPP.

**Comment by martinzink** [comment]
> Thanks for making the updates @martinzink.
> 
> Looking at the latest version, I think it would be better to introducing the new about pages and simply have two separate sections on the main MiNiFi page, one paragraph for Java, and one for CPP.

Not sure if I follow you here. :/ 
Should I remove the MiNiFi Java, MiNiFi C++ dropdowns and include those links in their about md. Which I should simple embed in the landing page?

**Comment by exceptionfactory** [comment]
> > Thanks for making the updates @martinzink.
> > Looking at the latest version, I think it would be better to introducing the new about pages and simply have two separate sections on the main MiNiFi page, one paragraph for Java, and one for CPP.
> 
> Not sure if I follow you here. :/ Should I remove the MiNiFi Java, MiNiFi C++ dropdowns and include those links in their about md. Which I should simple embed in the landing page?

Thanks for the reply, sorry for the confusion.

On further review, I see that the two about pages for Java and C++ are pulled in through an include to the main index.md. Although that content could be moved to the `partials` directory and then included, it seems like it would be best to simple copy the content from the about files into the index.md directly, avoiding the need for separate files. Is there some other reason for having that content in separate files, as opposed to just having it written directly in the index.md?

**Comment by martinzink** [comment]
> > > Thanks for making the updates @martinzink.
> > > Looking at the latest version, I think it would be better to introducing the new about pages and simply have two separate sections on the main MiNiFi page, one paragraph for Java, and one for CPP.
> > 
> > 
> > Not sure if I follow you here. :/ Should I remove the MiNiFi Java, MiNiFi C++ dropdowns and include those links in their about md. Which I should simple embed in the landing page?
> 
> Thanks for the reply, sorry for the confusion.
> 
> On further review, I see that the two about pages for Java and C++ are pulled in through an include to the main index.md. Although that content could be moved to the `partials` directory and then included, it seems like it would be best to simple copy the content from the about files into the index.md directly, avoiding the need for separate files. Is there some other reason for having that content in separate files, as opposed to just having it written directly in the index.md?

The reason I went with this convoluted way is that I wanted to create a proper up-to-date starting guide/landing page separately for java and c++ aswell and it would have made sense to include these small summaries there aswell.
After a little bit of time I felt that those pages are a bit out of the scope of this PR and wanna do them in a followup PR (maybe after the redesign?). 

But I agree its unneccessary complication for now, so I've removed these in https://github.com/apache/nifi-site/pull/79/commits/30521e9046501bb4ee55e36578343b47a92457b6 and I'll cross this bridge when I come to it :D

File: source/minifi/minifi-java-about.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+### MiNiFi Java
+The Java agent is built from the same codebase as NiFi, enabling it to run most of [NiFi's processors](http://nifi.apache.org/docs.html).
+However, it results in a larger binary distribution and consumes greater system resources, compared to MiNiFi C++.
```
Comment (szaszm) [inline source/minifi/minifi-java-about.md:3]: Another big thing for some people is the requirement to install Java on an edge device. Maybe we could add that to the list, if it's not too out-of-scope.
```suggestion
However, it results in a larger binary distribution, consumes greater system resources, and requires JRE, compared to MiNiFi C++.
```

File: source/minifi/minifi-java-guide-pushing-nifi-minifi.md (hunk 5 above)
Comment (martinzink) [reply source/minifi/minifi-java-guide-pushing-nifi-minifi.md]: good catch 👍 wasnt intentional fixed it in https://github.com/apache/nifi-site/pull/79/commits/db571ce47762a7ba99be83f7b45255bdb10fcbcb

File: layouts/partials/minifi-topbar.html [hunk 7]
```diff
@@ -16,30 +16,32 @@ <h1>
...
-                    <a href="#">Downloads</a>
+                    <a href="#">MiNiFi C++</a>
                     <ul class="dropdown">
-                        <li><a href="download.html">Download MiNiFi Components</a></li>
-                        <li><a href="https://cwiki.apache.org/confluence/display/MINIFI/Release+Notes"><i class="fa fa-external-link external-link"></i>MiNiFi Java Release Notes</a></li>
-                        <li><a href="https://cwiki.apache.org/confluence/display/MINIFI/Release+Notes#ReleaseNotes-MiNiFi(C++)"><i class="fa fa-external-link external-link"></i>MiNiFi C++ Release Notes</a></li>
+                        <li><a href="https://github.com/apache/nifi-minifi-cpp?tab=readme-ov-file#installing-as-a-service"><i class="fa fa-external-link external-link"></i>Installation</a></li>
+                        <li><a href="https://github.com/apache/nifi-minifi-cpp/blob/main/CONFIGURE.md"><i class="fa fa-external-link external-link"></i>Configuration</a></li>
+                        <li><a href="https://github.com/apache/nifi-minifi-cpp/blob/main/PROCESSORS.md"><i class="fa fa-external-link external-link"></i>Supported Processors</a></li>
```
Comment (exceptionfactory) [inline layouts/partials/minifi-topbar.html:32]: Recommend shortening this to just `Processors` as the label.
```suggestion
                        <li><a href="https://github.com/apache/nifi-minifi-cpp/blob/main/PROCESSORS.md"><i class="fa fa-external-link external-link"></i>Processors</a></li>
```

File: source/minifi/minifi-java-about.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+### MiNiFi Java
```
[the remark this answers is not in the document]
Comment (martinzink) [reply source/minifi/minifi-java-about.md]: It looks like I accidentally deleted the wrong file before, (removed the download.md and not the about.md) fixed in https://github.com/apache/nifi-site/pull/79/commits/48dba49cc028800271ae1eb86f0d90db726ac089

**Comment by exceptionfactory** [comment]
Thanks for the update @martinzink! I may have missed this earlier, but similar to the about pages, it seems better to keep one MiNiFi Download page, with separate sections for C++ and Java, similar to how it is now. This is the general pattern being considered now in the redesigned approach. Having fewer pages should be easier to maintain, and easier to navigate for someone looking to download sources or binaries. What do you think?

**Review by szaszm — APPROVED** [review]
No more comments from me, LGTM! Thanks @martinzink

**Comment by martinzink** [comment]
> Thanks for the update @martinzink! I may have missed this earlier, but similar to the about pages, it seems better to keep one MiNiFi Download page, with separate sections for C++ and Java, similar to how it is now. This is the general pattern being considered now in the redesigned approach. Having fewer pages should be easier to maintain, and easier to navigate for someone looking to download sources or binaries. What do you think?

The reason I split it into two because the topbar felt crammed with the new C++ and Java dropdowns so I removed the separate dropdown for Downloads, then I split it into two because it would be weird (for me at least) that the first download link from the C++ dropdown is actually for Java.

But I understand where you are coming from. What do you think would be the best, reintroduce the Download dropdown from the topbar and remove the links from the C++/Java dropdowns, or leave the download links in the C++/Java dropdowns but they would lead to the single download link?

**Comment by martinzink** [comment]
Thanks for quick and thorough reviews and merge @exceptionfactory @szaszm
I will keep an eye on the redesign and do any further improvements based on that.

**Pull request closed without merging**